sphinx/util/logging.py

```diff
@@ -27,3 +124,78 @@ def is_suppressed_warning(type, subtype, suppress_warnings):
                 return True
 
     return False
+
+
+class WarningSuppressor(logging.Filter):
+    """Filter logs by `suppress_warnings`."""
+
+    def __init__(self, app):
+        self.app = app
+        super(WarningSuppressor, self).__init__()
+
+    def filter(self, record):
+        type = getattr(record, 'type', None)
+        subtype = getattr(record, 'subtype', None)
+
+        if is_suppressed_warning(type, subtype, self.app.config.suppress_warnings):
+            return False
+        else:
+            self.app._warncount += 1
+            return True
+
+
+class WarningIsErrorFilter(logging.Filter):
+    """Raise exception if warning emitted."""
+
+    def __init__(self, app):
+        self.app = app
+        super(WarningIsErrorFilter, self).__init__()
+
+    def filter(self, record):
+        if self.app.warningiserror:
+            raise SphinxWarning(record.msg % record.args)
```

Comment:
I just pushed new `sphinx.util.logging` supporting INFO logs.
I hope you like it.